src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -191,6 +193,34 @@ private function registerCommonConfiguration(ContainerBuilder $container, array
         if ($config['name_converter']) {
             $container->setAlias('api_platform.name_converter', $config['name_converter']);
         }
+        $container->setParameter('api_platform.defaults', $this->normalizeDefaults($config['defaults'] ?? []));
+    }
+
+    private function normalizeDefaults(array $defaults): array
+    {
+        $normalizedDefaults = ['attributes' => []];
+        $reflectionClass = new \ReflectionClass(ApiResource::class);
```

Comment:
Maybe could we just hardcore the list of the 7 public properties instead of using the reflection and catching exceptions (which is known to be slow).

Reply:
Since it's executed at compile time I thought it was no big deal (and I'm not a big fan of hardcoding things) but I'm not against it.

Reply:
Wait, I actually need to fetch all properties from `ApiResource::class`, dynamically, which means Reflection is not an option, right? Or do you mean only refactoring the `$reflectionProperty->isPublic()` part?

Reply:
for instance you could add a `public static array` marked as `@internal` in `ApiResource` containing the list. It will also simplify the code.